### PR TITLE
Added SMTP test function in Alerting

### PR DESCRIPTION
**Pull request by atran-inverse**
# Description
Added a button to test SMTP configuration by sending a test email.

# Impacts
New button in Alerting page in Configuration

# Delete branch after merge
YES 

# NEWS file entries
* SMTP test function

### CODE REVIEW

**Comment by satkunas** [comment]
Error:

```
Error! SMTP Failed to connect to mail server: at /usr/share/perl5/vendor_perl/MIME/Lite.pm line 3036.
```

Using 
sogo.inverse.ca:443 w/ SSL

**Comment by atran-inverse** [comment]
Try port 587 and 465.

**Comment by julsemaan** [comment]
The message itself is good, but we'd just need to catch the error and die without the module name and line number (`at /usr/share/perl5/vendor_perl/MIME/Lite.pm line 3036.`)

File: html/pfappserver/lib/pfappserver/PacketFence/Controller/Configuration.pm [hunk 1]
```diff
@@ -546,6 +548,38 @@ sub all_subsections : Private {
...
+            Data => "PacketFence SMTP Test successful!\n"
+        );
+
+        my $results = eval {
+            pf::config::util::do_send_mime_lite($msg, %$alerting_config);
+        };
+        # the variable $@ holds the error
+        if ($@) {
+            $status = 400;
+            $status_msg = $@;
```
Comment (jrouzierinverse) [inline html/pfappserver/lib/pfappserver/PacketFence/Controller/Configuration.pm]: You can strip out the filename from the exception by using this function.

```perl
  $status_msg = pf::util::strip_filename_from_exceptions($@);
```

**Pull request merged**